coalib/bears/Bear.py

```diff
@@ -368,15 +370,35 @@ def download_cached_file(self, url, filename):
         :param filename: The filename it should get, e.g. "test.txt".
         :return:         A full path to the file ready for you to use!
         """
-        filename = join(self.data_dir, filename)
-        if exists(filename):
-            return filename
+        scrape = requests.head(url)
 
+        filename = join(self.data_dir, filename)
+        versionfile = join(self.data_dir, filename + 'version.txt')
```

Comment:
`.txt` isn't a good idea, as it can not be put in `.gitignore`.

`'.' + filename + '-version.etag'` ?